src/js/node/dgram.ts

```diff
@@ -1,23 +1,1082 @@
+    super(`The "${argName}" argument is required`);
+    this.code = "ERR_MISSING_ARGS";
+  }
+}
+
+class ERR_SOCKET_ALREADY_BOUND extends Error {
+  constructor() {
+    super("Socket is already bound");
+    this.code = "ERR_SOCKET_ALREADY_BOUND";
+  }
+}
+
+class ERR_SOCKET_BAD_BUFFER_SIZE extends Error {
+  constructor() {
+    super("Buffer size must be a number");
+    this.code = "ERR_SOCKET_BAD_BUFFER_SIZE";
+  }
+}
+
+class ERR_SOCKET_BUFFER_SIZE extends Error {
+  constructor(ctx) {
+    super(`Invalid buffer size: ${ctx}`);
+    this.code = "ERR_SOCKET_BUFFER_SIZE";
+  }
+}
+
+class ERR_SOCKET_DGRAM_IS_CONNECTED extends Error {
+  constructor() {
+    super("Socket is connected");
+    this.code = "ERR_SOCKET_DGRAM_IS_CONNECTED";
+  }
+}
+
+class ERR_SOCKET_DGRAM_NOT_CONNECTED extends Error {
+  constructor() {
+    super("Socket is not connected");
+    this.code = "ERR_SOCKET_DGRAM_NOT_CONNECTED";
+  }
+}
+
+class ERR_SOCKET_BAD_PORT extends Error {
+  constructor(name, port, allowZero) {
+    super(`Invalid ${name}: ${port}. Ports must be >= 0 and <= 65535. ${allowZero ? "0" : ""}`);
+    this.code = "ERR_SOCKET_BAD_PORT";
+  }
+}
+
+class ERR_SOCKET_DGRAM_NOT_RUNNING extends Error {
+  constructor() {
+    super("Socket is not running");
+    this.code = "ERR_SOCKET_DGRAM_NOT_RUNNING";
+  }
+}
+
+function isInt32(value) {
+  return value === (value | 0);
+}
+
+function validateAbortSignal(signal, name) {
+  if (signal !== undefined && (signal === null || typeof signal !== "object" || !("aborted" in signal))) {
+    throw new ERR_INVALID_ARG_TYPE(name, "AbortSignal", signal);
+  }
+}
+hideFromStack(validateAbortSignal);
+
+function validateString(value, name) {
+  if (typeof value !== "string") throw new ERR_INVALID_ARG_TYPE(name, "string", value);
+}
+hideFromStack(validateString);
+
+function validateNumber(value, name, min = undefined, max) {
+  if (typeof value !== "number") throw new ERR_INVALID_ARG_TYPE(name, "number", value);
+
+  if (
+    (min != null && value < min) ||
+    (max != null && value > max) ||
+    ((min != null || max != null) && NumberIsNaN(value))
+  ) {
+    throw new ERR_OUT_OF_RANGE(
+      name,
+      `${min != null ? `>= ${min}` : ""}${min != null && max != null ? " && " : ""}${max != null ? `<= ${max}` : ""}`,
+      value,
+    );
+  }
+}
+hideFromStack(validateNumber);
+
+function validatePort(port, name = "Port", allowZero = true) {
+  if (
+    (typeof port !== "number" && typeof port !== "string") ||
+    (typeof port === "string" && StringPrototypeTrim(port).length === 0) ||
+    +port !== +port >>> 0 ||
+    port > 0xffff ||
+    (port === 0 && !allowZero)
+  ) {
+    throw new ERR_SOCKET_BAD_PORT(name, port, allowZero);
+  }
+  return port | 0;
+}
+hideFromStack(validatePort);
+
+function validateFunction(value, name) {
+  if (typeof value !== "function") throw new ERR_INVALID_ARG_TYPE(name, "Function", value);
+}
+hideFromStack(validateFunction);
+
+const BIND_STATE_UNBOUND = 0;
```

Comment:
can you move these to the top of the file?

the bundler will be smart enough to inline all the primitive constants if they're at the top